### PR TITLE
Make provider code  bit more resilient to null values.

### PR DESCRIPTION
**Pull request by jacob314**
These exceptions don't impact the UI but are annoying when running with pause on exception.

### CODE REVIEW

**Comment by jacob314** [comment]
Fyi @rrousselGit

File: packages/devtools_app/lib/src/provider/provider_nodes.dart [hunk 1]
```diff
@@ -49,9 +49,13 @@ final _rawProviderIdsProvider =
       eval.getInstance(idRef, isAlive)
   ]);
 
-  return [
-    for (final idInstance in providerIdInstances) idInstance.valueAsString,
-  ];
+  final ret = <String>[];
+  for (final idInstance in providerIdInstances) {
+    if (idInstance != null) {
+      ret.add(idInstance.valueAsString);
```
Comment (rrousselGit) [inline packages/devtools_app/lib/src/provider/provider_nodes.dart:55]: Hum this is weird that "null" is received here

In which case does this happen?

File: packages/devtools_app/lib/src/provider/instance_viewer/instance_providers.dart [hunk 2]
```diff
@@ -360,6 +366,9 @@ final AutoDisposeFutureProviderFamily<InstanceDetails, InstancePath>
       if (enumDetails != null) return enumDetails;
 
       final classInstance = await eval.getClass(instance.classRef, isAlive);
+      if (classInstance == null) {
```
Comment (rrousselGit) [inline packages/devtools_app/lib/src/provider/instance_viewer/instance_providers.dart:369]: How is it possible that an object has no associated class?

File: packages/devtools_app/lib/src/provider/provider_nodes.dart [hunk 3]
```diff
@@ -49,9 +49,13 @@ final _rawProviderIdsProvider =
       eval.getInstance(idRef, isAlive)
   ]);
 
-  return [
-    for (final idInstance in providerIdInstances) idInstance.valueAsString,
-  ];
+  final ret = <String>[];
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/provider/provider_nodes.dart:52]: nit: can we have a more descriptive name here?

**Review by kenzieschmoll — APPROVED** [review]

File: packages/devtools_app/lib/src/provider/instance_viewer/instance_providers.dart (hunk 2 above)
Comment (jacob314) [reply packages/devtools_app/lib/src/provider/instance_viewer/instance_providers.dart:369]: if isAlive got disposed part way through, eval will short circuit with a null response.

File: packages/devtools_app/lib/src/provider/provider_nodes.dart (hunk 1 above)
Comment (jacob314) [reply packages/devtools_app/lib/src/provider/provider_nodes.dart:55]: Just using the https://github.com/flutter/samples/tree/master/provider_shopper
sample.

File: packages/devtools_app/lib/src/provider/instance_viewer/instance_providers.dart (hunk 2 above)
Comment (rrousselGit) [reply packages/devtools_app/lib/src/provider/instance_viewer/instance_providers.dart:369]: Oh I see. I forgot that only "safeEval" throws if the isAlive was disposed.

Throwing "CancelledException" should be ideal then. That won't cause undesired logs as it is filtered

Comment (rrousselGit) [reply packages/devtools_app/lib/src/provider/instance_viewer/instance_providers.dart:369]: Could you add a to-do that asks me to inspect all getClass/co calls to spot issues?

I doubt this is the only case.

**Comment by Hixie** [comment]
What is the status of this PR? It seems to have been abandoned?

**Pull request closed without merging**